sycl/test-e2e/Matrix/joint_matrix_apply_two_matrices_impl.hpp

```diff
@@ -113,6 +113,11 @@ bool test() {
   else if constexpr (std::is_same_v<Ta, int8_t>)
     std::cout << "int8_t " << TM << "x" << TN << "x" << TK << ": "
               << (res ? "passed" : "failed") << std::endl;
+  free(C, q);
+  free(D, q);
+  free(A, q);
+  free(Ar, q);
+
   return res;
```

Comment:
This change is unrelated to OOB but is causing memory corruption on some machines so I added it as part of this PR.